main.go

```diff
@@ -113,51 +112,27 @@ func buildRegistryArguments(argPrefix string, argDescription string) RepositoryA
 	}
 }
 
-func connectToRegistry(args RepositoryArguments) (*registry.Registry, error) {
-	origUrl := *args.RegistryURL
-	url := origUrl
-	username := ""
-	password := ""
+func connectToRegistry(args RepositoryArguments, auths map[string]docker.AuthConfiguration) (*registry.Registry, error) {
 
-	if strings.HasPrefix(url, "ecr:") {
-		registryId := strings.TrimPrefix(url, "ecr:")
+	url := *args.RegistryURL
+	urlWithoutPrefix := strings.TrimPrefix(url, "https://")
 
-		sess, err := session.NewSession()
-		if err != nil {
-			return nil, fmt.Errorf("Failed to create new AWS SDK session. %v", err)
-		}
-		svc := ecr.New(sess)
-		params := &ecr.GetAuthorizationTokenInput{
-			RegistryIds: []*string{
-				aws.String(registryId), // Required
-			},
-		}
-
-		resp, err := svc.GetAuthorizationToken(params)
-		if err != nil {
-			return nil, fmt.Errorf("Failed to get ECR authorization token for registry %s. %v", registryId, err)
-		}
-
-		decoded, err := base64.StdEncoding.DecodeString(*resp.AuthorizationData[0].AuthorizationToken)
-		if err != nil {
-			return nil, fmt.Errorf("Failed to decode base64 encoded authorization data for ECR registry %s. %v", registryId, err)
-		}
-
-		parts := strings.Split(string(decoded), ":")
+	username := ""
+	password := ""
 
-		url = *resp.AuthorizationData[0].ProxyEndpoint
-		username = parts[0]
-		password = parts[1]
+	if auth, ok := auths[urlWithoutPrefix]; ok {
+		username = auth.Username
+		password = auth.Password
 	}
 
-	registry, err := registry.New(url, username, password)
+	registry, err := registry.NewInsecure(url, username, password)
```

Comment:
Switching to allow insecure connections seems like a dangerous thing to me. Can this be switched back to `New` instead?